server/sources.py

```diff
@@ -0,0 +1,302 @@
+            if self.net1 is None or self.net2 is None:
+                raise ValueError(f"must specify net1 and net2 in order to use different 'by' values (current: {self.by}, given: {by})")
+
+            net1_vs = self.net1.igraph.vs
+            net2_vs = self.net2.igraph.vs
+
+            if self_by == 'index':
+                for p1_id, p2_id, score in self.tricol:
+                    yield net1_vs[p1_id], net2_vs[p2_id], score
+            else:
+                for p1_by, p2_by, score in self.tricol:
+                    p1s = net1_vs.select(**{self_by:p1_by})
+                    p2s = net2_vs.select(**{self_by:p2_by})
+                    if len(p1s) > 0 and len(p2s) > 0:
+                        yield p1s[0], p2s[0], score
+        elif by == 'index':
+            for p1, p2, score in self.iter_tricol(by='object'):
+                yield p1.index, p2.index, score
+        else:
+            for p1, p2, score in self.iter_tricol(by='object'):
+                yield p1[by], p2[by], score
+
+
+def read_tricol_scores(file_path, net1=None, net2=None, by='name', **kwargs):
+    if 'delimiter' not in kwargs:
+        kwargs['delimiter'] = '\t'
+    return TricolScoreMatrix(util.iter_csv(file_path, **kwargs), net1=net1, net2=net2, by=by)
+
+def identity_scores(net, by='name'):
+    return TricolScoreMatrix([(v, v, 1) for v in net.iter_vertices(by=by)], by=by)
+
+
+class IsobaseLocal(object):
+    def __init__(self, base_path):
+        self.base_path = base_path
+
+    def _check_valid_species(self, species_name):
+        if not species_name.isalnum():
+            raise ValueError(f'invalid species name: {species_name}')
+
+    def get_network(self, species_name):
+        self._check_valid_species(species_name)
+        species_path = path.join(self.base_path, f'{species_name}.tab')
+
+        if not path.isfile(species_path):
+            raise IOError(f'network file for {species_name} was not found')
+
+        return read_net_tsv_edgelist(species_path)
+
+    def get_score_matrix(self, species1_name, species2_name=None, net1=None, net2=None):
+        self._check_valid_species(species1_name)
+
+        if species2_name is None:
+            matrix_path = path.join(self.base_path, f'{species1_name}-blast.tab')
+
+            if not path.isfile(matrix_path):
+                raise IOError(f'score matrix file for {species1_name} was not found')
+        else:
+            self._check_valid_species(species2_name)
+            matrix_path = path.join(self.base_path, f'{species1_name}-{species2_name}-blast.tab')
+
+            if not path.isfile(matrix_path):
+                raise IOError(f'score matrix file for {species1_name}-{species2_name} was not found')
+
+        return read_tricol_scores(matrix_path, net1=net1, net2=net2)
+
+    def get_ontology_mapping(self, networks=None):
+        with open(path.join(self.base_path, 'go.json'), 'r') as go_f:
+            return json.load(go_f)
+
+class StringDB(object):
```

Comment:
We must decide where this database will be. Now we have two options:
1.- Dockerize this database and the default host won't be localhost (will be the container name)
2.- Don't dockerize it and open the port in ther server, so we need to access it with an external ip

Reply:
I vote for 1 as it can be changed to 2 quite easily if needed. I already have some scripts to import the DB, create some indexes and perform some general cleanup. A postgres base should be enough for the image.

Reply:
I prefer 1 too